### PR TITLE
(GAP-1984) Adds Navbar for authenticated users

### PR DESCRIPTION
**Pull request by john-tco**
## Description

[GAP-1984](https://technologyprogramme.atlassian.net/jira/software/projects/GAP/boards/216?selectedIssue=GAP-1984)

todo - drill the `isUserLoggedIn` prop into every single applicant page OR remove `Layout` from every single applicant page and render `Layout` jsx from app entrypoint

## Type of change

Please check the relevant options.

- [ ] Bug fix (non-breaking change which fixes an issue).
- [ ] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).
- [ ] This change requires a documentation update.

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes:

- [ ] Unit Test

- [ ] Integration Test (if applicable)

- [ ] End to End Test (if applicable)

## Screenshots (if appropriate):

Please attach screenshots of the change if it is a UI change:

# Checklist:

- [ ] If I have listed depedenencies above, I have ensured that they are present in the target branch.
- [ ] I have performed a self-review of my code.
- [ ] I have commented my code in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation where applicable.
- [ ] I have ran cypress tests and they all pass locally.

[GAP-1984]: https://technologyprogramme.atlassian.net/browse/GAP-1984?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/admin/src/components/layout/navigation/index.ts
Comment (dominicwest) [inline packages/admin/src/components/layout/navigation/index.ts:1]: This file needed?

File: packages/admin/src/pages/_app.page.js [hunk 1]
```diff
@@ -48,9 +54,12 @@ const MyApp = ({ Component, pageProps, cookies }) => {
 
 MyApp.getInitialProps = async (appContext) => {
   const appProps = await App.getInitialProps(appContext);
+  const { req } = appContext.ctx;
+  const sessionToken = req.cookies[SESSION_COOKIE_NAME];
+  const isValidAdminSession = await isAdminSessionValid(sessionToken);
   const cookies =
     typeof window === 'undefined' ? {} : appContext.ctx.req.cookies;
-  return { ...appProps, cookies };
+  return { ...appProps, cookies, isUserLoggedIn: isValidAdminSession };
```
Comment (dominicwest) [inline packages/admin/src/pages/_app.page.js:62]: Nice, glad the admin app at least can be done easily 😭

File: packages/admin/src/components/layout/navigation/MobileNavigationBar.tsx
Comment (dominicwest) [inline packages/admin/src/components/layout/navigation/MobileNavigationBar.tsx:1]: Consider shifting to gap-web-ui if there's enough similarities between admin & applicant

File: packages/applicant/src/pages/api/logout.page.tsx
Comment (dominicwest) [inline packages/applicant/src/pages/api/logout.page.tsx:1]: Do we need similar changes in the admin logout api page?

File: packages/applicant/src/pages/applications/index.page.tsx [hunk 2]
```diff
@@ -21,15 +20,19 @@ export const getServerSideProps: GetServerSideProps<ApplicationsPage> = async ({
   };
 };
 
-const ExistingApplications = ({ applicationData }: ApplicationsPage) => {
+const ExistingApplications = ({
+  applicationData,
+  isUserLoggedIn,
+}: ApplicationsPage & {
+  isUserLoggedIn: boolean;
```
Comment (dominicwest) [inline packages/applicant/src/pages/applications/index.page.tsx:27]: Would declare this as a well named type that we can reuse. Something like `GetInitialPropsType`

Comment (john-tco) [reply packages/applicant/src/pages/applications/index.page.tsx:27]: Good shout :)

File: packages/admin/src/components/layout/navigation/index.ts
Comment (john-tco) [reply packages/admin/src/components/layout/navigation/index.ts:1]: Yes

File: packages/admin/src/components/layout/navigation/MobileNavigationBar.tsx
Comment (john-tco) [reply packages/admin/src/components/layout/navigation/MobileNavigationBar.tsx:1]: I will do it

File: packages/applicant/src/pages/api/logout.page.tsx [hunk 3]
```diff
@@ -1,12 +1,33 @@
...
-  req: NextApiRequest,
-  res: NextApiResponse
-) {
-  res.redirect(
+const Logout = async (req: NextApiRequest, res: NextApiResponse) => {
+  const sessionCookie = getSessionIdFromCookies(req);
+  if (sessionCookie) await logoutAdmin(sessionCookie);
+  res.setHeader(
+    'Set-Cookie',
+    `session_id=deleted; Path=/; secure; HttpOnly; SameSite=Strict; expires=Thu, 01 Jan 2003 00:00:00 GMT`
```
Comment (jgunnCO) [inline packages/applicant/src/pages/api/logout.page.tsx:10]: same as the other PR - do you not want to delete `user-service-token` every time and then delete `session_id` only when present?

File: packages/applicant/src/utils/session.ts [hunk 4]
```diff
@@ -0,0 +1,16 @@
+import { NextApiRequest } from 'next';
+import { NextApiRequestCookies } from 'next/dist/server/api-utils';
+import { NextIncomingMessage } from 'next/dist/server/request-meta';
+
+const getSessionIdFromCookies = (
+  req:
+    | (NextIncomingMessage & { cookies: NextApiRequestCookies })
+    | NextApiRequest
+) => {
+  if (!process.env.SESSION_COOKIE_NAME) {
```
Comment (jgunnCO) [inline packages/applicant/src/utils/session.ts:10]: add to `.env.example`

File: packages/applicant/src/pages/api/logout.page.tsx [hunk 5]
```diff
@@ -1,12 +1,33 @@
...
+const getAxiosSessionConfig = (sessionId: string) => ({
+  withCredentials: true,
+  headers: {
+    Cookie: `SESSION=${sessionId};`,
+  },
+});
+
+const logoutAdmin = async (sessionCookie: string) =>
+  axios.delete(
+    `${process.env.ADMIN_BACKEND_HOST}/logout`,
```
Comment (jgunnCO) [inline packages/applicant/src/pages/api/logout.page.tsx:29]: need to add this to `.env.example` as well

**Review by jgunnCO — APPROVED** [review]

**Pull request merged**